File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        const int n = nums.size();
        vector<vector<int>> count_arr(n+1);
        unordered_map<int,int> count_map;

        for (int num: nums) {
            count_map[num]++;
        }

        for (auto& c: count_map) {
            count_arr[c.second].push_back(c.first);
        }
        vector<int> output;
        int limit_k = 0;
        for (int i=n; i>=0 && limit_k<k; i--) {
            for (int num:count_arr[i]) {
                output.push_back(num);
                limit_k++;
                if (limit_k == k) break;
            }
        }
        return output;
    }


};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = defaultdict(int)
        for num in nums:
            count[num] += 1
        freq_list = []
        for num, cnt in count.items():
            freq_list.append([cnt, num])
        freq_list.sort()

        res = []
        while len(res)<k:
            print(res)
            res.append(freq_list.pop()[1])

        return res